SkylineToolsStore/src/org/labkey/skylinetoolsstore/view/SkylineToolDetails.jsp

```diff
@@ -23,16 +23,35 @@
 <%@ page import="java.util.Map" %>
 <%@ page import="java.util.Arrays" %>
 <%@ page import="org.labkey.api.portal.ProjectUrls" %>
+<%@ page import="org.labkey.api.view.template.ClientDependencies" %>
+<%@ page import="org.labkey.api.settings.AppProps" %>
+<%@ page import="org.labkey.api.util.DOM" %>
+<%@ page import="static org.labkey.api.util.DOM.IMG" %>
+<%@ page import="static org.labkey.api.util.DOM.Attribute.src" %>
+<%@ page import="static org.labkey.api.util.DOM.Attribute.alt" %>
+<%@ page import="org.labkey.api.util.PageFlowUtil" %>
 <%@ page extends="org.labkey.api.jsp.JspBase" %>
+
+<%!
+    @Override
+    public void addClientDependencies(ClientDependencies dependencies)
+    {
+        dependencies.add("internal/jQuery");
+        dependencies.add("skylinetoolsstore/js/functions.js");
+    }
+
+    public final HtmlString editIconImgHtml = HtmlString.unsafe(IMG(DOM.at(src, getWebappURL("skylinetoolsstore/img/pencil.png")).at(alt, "Pencil")).renderToString());
```

Comment:
@labkey-matthewb @labkey-adam this works but routing this through `unsafe` feels unsatisfying. Do you have a recommendation for a different pattern that works with `link()`?

Reply:
I'm afraid not. `LinkBuilder` and `link()` should take `Renderable`, which would avoid `HtmlString` entirely. Feel free to open an issue to me so I don't forget...

Reply:
Actually, `DOM.createHtmlFragment()` is your friend here; just don't look at the impl. 😄 I also have a local change to switch html-taking methods to `Renderable`.

Reply:
https://github.com/LabKey/platform/pull/5406

Reply:
Thanks. My PR is targeting 24.3 so the changes to builders accepting `Renderable` will not be available.  Should I target "develop" instead?  For now, I have replaced `HtmlString.unsafe()` with `DOM.createHtml()` to make it look less ugly.

Reply:
Whatever's easiest for you... I'd probably stick with 24.3 since that's what you have and my PR isn't merged yet. Though note that develop is fine for all the strict CSP work.